shared_modules/soundzone_protocol/soundzone_protocol.py

```diff
@@ -0,0 +1,140 @@
+
+    Methods
+    -------
+    _add_command_to_command()
+        Used to assign an command object corresponding the command name.
+    add_command()
+        Used to add a command to to #command.
+    encode()
+        Encodes the class into a hex string.
+    decode()
+        Decodes a buffer Hex string following SoundZone Protocol
+    """
+    def __init__(self):
+        """
+        Initiates cid, cid_name and payload
+        """
+        self.cid = None
+        self.cid_name = None
+        self.command = None
+
+    def _add_command_to_command(self):
+        """
+        Used to assign a command object corresponding the command name to #command.
+        :return: None
+        """
+        if self.cid_name == "checkConn":
+            command = XF1CheckConn()
+        elif self.cid_name == "send":
+            command = X01Send()
+        elif self.cid_name == "enroll":
+            command = XA1Enroll()
+        elif self.cid_name == "set_sound_format":
+            command = XB3SetSoundFormat()
+        else:
+            raise Exception(f"Command name not known, got: {self.cid_name}")
+        self.command = command
+
+    def add_command(self, command: [int, str]):
+        """
+        Used to add a command to to #command.
+        :param command: The cid or cid_name used to find the correct command obj.
+        :return: None
+        """
+        if type(command) is int:
+            if command not in _supported_cids.keys():
+                raise Exception(f"CID not supported, got: {hex(command)}")
+            self.cid = command
+            self.cid_name = _supported_cids[command]
+        elif type(command) is str:
+            if command not in _supported_cids.values():
+                raise Exception(f"Command name not supported, got: {hex(command)}")
+            self.cid = _get_key(command, _supported_cids)
+            self.cid_name = command
+        else:
+            raise Exception(f"Command has to be of type str or int, got {type(command)}")
+        self._add_command_to_command()
+
+    def encode(self):
+        """
+        Encodes the class into a hex string.
+        :return: hex string, encoded as described in SoundZone Protocol
+        """
+        encoded_apl = "{:02X}".format(self.cid)
+        encoded_apl += self.command.encode()
+        return encoded_apl
+
+    def decode(self, buffer):
+        """
+        Decodes a buffer Hex string following SoundZone Protocol
+        :param buffer: Hex string to decode
+        :return: None
+        """
+        buffer = [int(x+y, 16) for x, y in zip(buffer[0::2], buffer[1::2])]
+
+        self.cid = buffer[0]
+        payload = buffer[1:]
+
+        if self.cid not in _supported_cids.keys():
+            raise Exception(f"CID not supported, got: {hex(self.cid)}")
+
+        self.cid_name = _supported_cids[self.cid]
+        self._add_command_to_command()
+        self.command.decode(payload)
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
Hvorfor encodes alt data, er det ikke spild af krafter? Hvis vi har defineret en CID er det så ikke bare at ud fra header ved den hvordan den skal læse dataet?

F.eks. Hvis CID = set_sound_format
Så første 2 bytes er til CID, næste 2 er for sample rate og næste 2 til resolution osv. 
For hvis det gøres på den her måde, så vil selve musiken der sendes vel også skulle encodes og decodes?